In [12]:
## Importing the necessary liabraries
import selenium
import pandas as pd
from selenium import webdriver

### Q1: Write a python program to scrape data for “Data Analyst” Job position in “Bangalore” location.<br> 
You have to scrape the job-title, job-location, company_name,experience_required. You have to scrape first 10 jobs data.<br>
This task will be done in following steps:
1. first get the webpage https://www.naukri.com/
2. Enter “Data Analyst” in “Skill,Designations,Companies” field and enter “Bangalore” 
in “enter the location” field.
3. Then click the search button.
4. Then scrape the data for the first 10 jobs results you get.
5. Finally create a dataframe of the scraped data.
Note- All of the above steps have to be done in code. No step is to be done manually.


In [292]:
driver=webdriver.Edge(r'C:\Users\Mandar Shinde\Desktop\DataScience\DATA TRAINED\FlipRobo Internship\msedgedriver.exe')

In [4]:
##Getting the naukari.com home page
url='https://www.naukri.com/'
driver.get(url)

In [5]:
## Accessing the job search bar
search_job=driver.find_element_by_xpath('//input[@class="sugInp"]')
search_job

<selenium.webdriver.remote.webelement.WebElement (session="9e44d615dd20b2570bdca60d1079a858", element="5751bef5-e6d6-4c18-8e62-fd6926aae69d")>

In [6]:
## Entering 'Data Analyst' in job search bar
search_job.send_keys('Data Analyst')

In [7]:
## Accessing the location search bar
search_loc=driver.find_element_by_id("qsb-location-sugg")
search_loc

<selenium.webdriver.remote.webelement.WebElement (session="9e44d615dd20b2570bdca60d1079a858", element="32102363-4682-48d4-992f-c6d9ab8dbf82")>

In [8]:
## Entering 'Bangalore' in location search bar
search_loc.send_keys('Bangalore')

In [9]:
## Accessing and clicking the search button
search_btn=driver.find_element_by_xpath('//button[@class="btn"]')

In [10]:
search_btn.click()

In [11]:
## extracting tags having job titles
job_tags=driver.find_elements_by_xpath('//a[@class="title fw500 ellipsis"]')
job_tags

[<selenium.webdriver.remote.webelement.WebElement (session="9e44d615dd20b2570bdca60d1079a858", element="1610d08c-8fdf-4848-b4a1-c3fb8cdb3c5c")>,
 <selenium.webdriver.remote.webelement.WebElement (session="9e44d615dd20b2570bdca60d1079a858", element="e5b1bf9e-14dc-4ed0-b0fc-fbf5e6cd16d7")>,
 <selenium.webdriver.remote.webelement.WebElement (session="9e44d615dd20b2570bdca60d1079a858", element="c051eaad-033b-4136-a822-313d157ad2dd")>,
 <selenium.webdriver.remote.webelement.WebElement (session="9e44d615dd20b2570bdca60d1079a858", element="ae85c3a3-6b30-4f71-abba-1118fe13e049")>,
 <selenium.webdriver.remote.webelement.WebElement (session="9e44d615dd20b2570bdca60d1079a858", element="9fca339d-af70-4ce1-97ca-7c9c27eeea5a")>,
 <selenium.webdriver.remote.webelement.WebElement (session="9e44d615dd20b2570bdca60d1079a858", element="752dfa68-c00a-480c-a2ab-cac5f4de22db")>,
 <selenium.webdriver.remote.webelement.WebElement (session="9e44d615dd20b2570bdca60d1079a858", element="b87c829e-6121-48c6-94a8-98

In [14]:
## saving the text in a list
job_titles=[]
for i in job_tags:
    job_titles.append(i.text)
job_titles[:10]

['Data Analyst',
 'Data Analyst',
 'Data Analyst',
 'Product/Data Analyst - SQL/Tableau/Qlikview',
 'Process Data Analyst',
 'Data Analyst',
 'Data Analyst',
 'Product Data Analyst - IT',
 'Data Analyst II - Customer Support',
 'Data Analyst']

In [18]:
## extracting tags having locations and saving the text in a list
loc_tags=driver.find_elements_by_xpath('//li[@class="fleft grey-text br2 placeHolderLi location"]')
locations=[]
for i in loc_tags:
    locations.append(i.text)
locations[:10]

['Bangalore/Bengaluru',
 'Bangalore/Bengaluru',
 'Bangalore/Bengaluru',
 'Bangalore/Bengaluru',
 'Bangalore/Bengaluru',
 'Bangalore/Bengaluru',
 'Bangalore/Bengaluru',
 'Bangalore/Bengaluru',
 'Bangalore/Bengaluru',
 'Bangalore/Bengaluru']

In [21]:
## extracting all the tags having company names and saving the text in a list
company_tags=driver.find_elements_by_xpath('//a[@class="subTitle ellipsis fleft"]')
company_names=[]
for i in company_tags:
    company_names.append(i.text)
company_names[:10]

['Innovsource Services Private Limited',
 'Rapido -bike taxi',
 'ICF Next',
 'Connexions',
 'Hitachi ABB Power Grids',
 'Rapido -bike taxi',
 'SORTING HAT TECHNOLOGIES PVT LTD',
 'GAMA India',
 'VIMEO TECHNOLOGIES PRIVATE LIMITED',
 'PharmEasy']

In [22]:
## extracting all the tags having experience and saving the text in a list
exp_tags=driver.find_elements_by_xpath('//li[@class="fleft grey-text br2 placeHolderLi experience"]')
experience=[]
for i in exp_tags:
    experience.append(i.text)
experience[:10]

['1-6 Yrs',
 '2-4 Yrs',
 '1-2 Yrs',
 '1-4 Yrs',
 '2-6 Yrs',
 '2-4 Yrs',
 '2-5 Yrs',
 '2-4 Yrs',
 '1-5 Yrs',
 '1-3 Yrs']

In [23]:
## Creating a dataframe to store data for 'Data Analyst' jobs in 'Bangalore' 
DataAnalystJobs=pd.DataFrame()

DataAnalystJobs['Title']=job_titles[:10]
DataAnalystJobs['Company']=company_names[:10]
DataAnalystJobs['Experience']=experience[:10]
DataAnalystJobs['Location']=locations[:10]

DataAnalystJobs

Title  \
0                                 Data Analyst   
1                                 Data Analyst   
2                                 Data Analyst   
3  Product/Data Analyst - SQL/Tableau/Qlikview   
4                         Process Data Analyst   
5                                 Data Analyst   
6                                 Data Analyst   
7                    Product Data Analyst - IT   
8           Data Analyst II - Customer Support   
9                                 Data Analyst   

                                Company Experience             Location  
0  Innovsource Services Private Limited    1-6 Yrs  Bangalore/Bengaluru  
1                     Rapido -bike taxi    2-4 Yrs  Bangalore/Bengaluru  
2                              ICF Next    1-2 Yrs  Bangalore/Bengaluru  
3                            Connexions    1-4 Yrs  Bangalore/Bengaluru  
4               Hitachi ABB Power Grids    2-6 Yrs  Bangalore/Bengaluru  
5                     Rapido -bike taxi    2-4 Yrs  Bangalore/Bengaluru  
6      SORTING HAT TECHNOLOGIES PVT LTD    2-5 Yrs  Bangalore/Bengaluru  
7                            GAMA India    2-4 Yrs  Bangalore/Bengaluru  
8    VIMEO TECHNOLOGIES PRIVATE LIMITED    1-5 Yrs  Bangalore/Bengaluru  
9                             PharmEasy    1-3 Yrs  Bangalore/Bengaluru

### Q2: Write a python program to scrape data for “Data Scientist” Job position in “Bangalore” location. <br> 
You have to scrape the job-title, job-location,company_name, full job-description. You have to scrape first 10 jobs data.<br>
This task will be done in following steps:
1. first get the webpage https://www.naukri.com/
2. Enter “Data Scientist” in “Skill,Designations,Companies” field and enter “Bangalore” in “enter the location” field.
3. Then click the search button.
4. Then scrape the data for the first 10 jobs results you get.
5. Finally create a dataframe of the scraped data.<br>
Note- 
1. All of the above steps have to be done in code. No step is to be done manually
2. Please note that you have to scrape full job description. For that you may have to open each job separately as shown below

In [51]:
##Getting the naukari.com home page
url='https://www.naukri.com/'
driver.get(url)

In [52]:
## accessing the search bar
search_job=driver.find_element_by_xpath('//input[@class="sugInp"]')
search_job

<selenium.webdriver.remote.webelement.WebElement (session="19a8b95142d5dec518a61d93002fa57d", element="c3f7cb8b-a48c-474b-9b4c-06dc191361dd")>

In [53]:
## Entering 'Data Scientist' in search bar
search_job.send_keys('Data Scientist')

In [54]:
## Accessing location search bar
search_loc=driver.find_element_by_id("qsb-location-sugg")
search_loc

<selenium.webdriver.remote.webelement.WebElement (session="19a8b95142d5dec518a61d93002fa57d", element="1385a8c1-088f-42ba-89de-9d12af8d864e")>

In [55]:
## Entering 'Bangalore' in location search bar
search_loc.send_keys('Bangalore')

In [56]:
## Accessing and clicking the search button
search_btn=driver.find_element_by_xpath('//button[@class="btn"]')

In [57]:
search_btn.click()

In [33]:
## extracting tags having job titles
job_tags=driver.find_elements_by_xpath('//a[@class="title fw500 ellipsis"]')
job_titles=[]
for i in job_tags:
    job_titles.append(i.text)
job_titles[:10]

['Senior Data Scientist',
 'Senior Data Scientist',
 'Senior Data Scientist',
 'Senior Data Scientist',
 'DATA SCIENTIST',
 'Sr. Data Scientist / Tech Lead - Data Science',
 'Senior Data Scientist',
 'Data Engineers , Data Scientist and SAP',
 'Python - Data Scientist',
 'Western Digital Data Scientist 3']

In [34]:
## extracting tags having locations and saving the text in a list
loc_tags=driver.find_elements_by_xpath('//li[@class="fleft grey-text br2 placeHolderLi location"]')
locations=[]
for i in loc_tags:
    locations.append(i.text)
locations[:10]

['Bangalore/Bengaluru',
 'Bangalore/Bengaluru',
 'Bangalore/Bengaluru',
 'New Delhi, Bangalore/Bengaluru, Delhi / NCR\n(WFH during Covid)',
 'Bangalore/Bengaluru',
 'Bangalore/Bengaluru',
 'Bangalore/Bengaluru',
 'Bangalore/Bengaluru',
 'Bangalore/Bengaluru',
 'Bangalore/Bengaluru']

In [35]:
## extracting all the tags having company names and saving the text in a list
company_tags=driver.find_elements_by_xpath('//a[@class="subTitle ellipsis fleft"]')
company_names=[]
for i in company_tags:
    company_names.append(i.text)
company_names[:10]

['Datamatics Global Services Ltd',
 'Publicis Groupe',
 'Bidgely Technologies Private Limited',
 'Goals 101 Data Solutions Pvt. Ltd.',
 'McAfee Software (India) Pvt. Ltd',
 'Confidential',
 'RANDSTAD INDIA PVT LTD',
 'Teamware Solutions',
 'Mindtree Limited',
 'Western Digital']

In [58]:
urls=driver.find_elements_by_xpath('//a[@class="title fw500 ellipsis"]')

In [59]:
job_opening_urls=[]
for i in urls:
    job_opening_urls.append(i.get_attribute('href'))
job_opening_urls

['https://www.naukri.com/job-listings-senior-data-scientist-datamatics-global-services-ltd-bangalore-bengaluru-8-to-13-years-040821003517?src=jobsearchDesk&sid=16281036436188096&xp=1&px=1',
 'https://www.naukri.com/job-listings-senior-data-scientist-publicis-groupe-bangalore-bengaluru-2-to-5-years-030821501259?src=jobsearchDesk&sid=16281036436188096&xp=2&px=1',
 'https://www.naukri.com/job-listings-senior-data-scientist-bidgely-technologies-private-limited-bangalore-bengaluru-4-to-6-years-030821500043?src=jobsearchDesk&sid=16281036436188096&xp=3&px=1',
 'https://www.naukri.com/job-listings-senior-data-scientist-goals-101-data-solutions-pvt-ltd-new-delhi-bangalore-bengaluru-delhi-ncr-4-to-8-years-030821000870?src=jobsearchDesk&sid=16281036436188096&xp=4&px=1',
 'https://www.naukri.com/job-listings-data-scientist-mcafee-software-india-pvt-ltd-bangalore-bengaluru-4-to-8-years-040821500539?src=jobsearchDesk&sid=16281036436188096&xp=5&px=1',
 'https://www.naukri.com/job-listings-sr-data-sci

In [64]:
## Scraping the detailed job descriptions for 10 jobs and saving them in a list
jd=[]

for i in job_opening_urls:
    driver.get(i)
    try:
        jd_tags=driver.find_element_by_xpath('//div[@class="dang-inner-html"]')
        jd.append(jd_tags.text.replace('\n',''))
    except:
        jd.append('---')
jd

['Roles and ResponsibilitiesDetails in lined for Hiring Team to focus:The hiring is for a single Data Scientist (7-8+yrs) as he/she shall be an individual contributor.We shall seek for a strong java/python/scala with expertise into frameworks ML (spark/keras/tensorflow).ML concepts has to be thorough with proven expertise in ( Decision Trees, Random Forest, NLP) Deep Learning is excluded.Desired Candidate ProfilePerks and Benefits',
 'Work location: Bengaluru (Currently, Remote)Shift timing: 12PM to 9PMWork with stakeholders throughout the organization to identify opportunities for leveraging company data to drive business solutions.Develop a use case roadmap for a problem area or capability for the business. Frame the business problem into a Data Science or modelling problem.Extract data from multiple sources. Mine and analyze data from company databases to drive optimization and improvement of product.Work as the data strategist, identifying and integrating new datasets that can be l

In [63]:
## Creating a dataframe to store data for 'Data Analyst' jobs in 'Bangalore' 
DataScientistJobs=pd.DataFrame()

DataScientistJobs['Title']=job_titles[:10]
DataScientistJobs['Company']=company_names[:10]
DataScientistJobs['Location']=locations[:10]
DataScientistJobs['Job Description']=jd[:10]

DataScientistJobs

Title  \
0                          Senior Data Scientist   
1                          Senior Data Scientist   
2                          Senior Data Scientist   
3                          Senior Data Scientist   
4                                 DATA SCIENTIST   
5  Sr. Data Scientist / Tech Lead - Data Science   
6                          Senior Data Scientist   
7        Data Engineers , Data Scientist and SAP   
8                        Python - Data Scientist   
9               Western Digital Data Scientist 3   

                                Company  \
0        Datamatics Global Services Ltd   
1                       Publicis Groupe   
2  Bidgely Technologies Private Limited   
3    Goals 101 Data Solutions Pvt. Ltd.   
4      McAfee Software (India) Pvt. Ltd   
5                          Confidential   
6                RANDSTAD INDIA PVT LTD   
7                    Teamware Solutions   
8                      Mindtree Limited   
9                       Western Digital   

                                            Location  \
0                                Bangalore/Bengaluru   
1                                Bangalore/Bengaluru   
2                                Bangalore/Bengaluru   
3  New Delhi, Bangalore/Bengaluru, Delhi / NCR\n(...   
4                                Bangalore/Bengaluru   
5                                Bangalore/Bengaluru   
6                                Bangalore/Bengaluru   
7                                Bangalore/Bengaluru   
8                                Bangalore/Bengaluru   
9                                Bangalore/Bengaluru   

                                     Job Description  
0  Roles and ResponsibilitiesDetails in lined for...  
1  Work location: Bengaluru (Currently, Remote)Sh...  
2  Bidgely is looking for extraordinary and dynam...  
3  Roles and Responsibilities:Work as part of a p...  
4  Role Overview:Data scientist will be doing res...  
5  Role : Sr Data Scientist / Tech Lead - Data Sc...  
6  Roles and ResponsibilitiesJob DescriptionWhat ...  
7  Dynamic recruiting professional with 7 - 10 ye...  
8  Basic Qualifications3-5 years of hands-on expe...  
9  Design, develop and maintain Machine Learning/...

### Q3: In this question you have to scrape data using the filters available on the 
webpage as shown below:
You have to use the location and salary filter.<br>
You have to scrape data for “Data Scientist” designation for first 10 job results.<br>
You have to scrape the job-title, job-location, company_name,experience_required.<br>
The location filter to be used is “Delhi/NCR”<br>
The salary filter to be used is “3-6” lakhs<br>
The task will be done as shown in the below steps:
1. first get the webpage https://www.naukri.com/
2. Enter “Data Scientist” in “Skill,Designations,Companies” field .
3. Then click the search button.
4. Then apply the location filter and salary filter by checking the respective boxes
4. Then scrape the data for the first 10 jobs results you get.
5. Finally create a dataframe of the scraped data.
Note- All of the above steps have to be done in code. No step is to be done 
manually.

In [87]:
##Getting the naukari.com home page
url='https://www.naukri.com/'
driver.get(url)

In [88]:
## Accessing the search bar
search_job=driver.find_element_by_xpath('//input[@class="sugInp"]')
search_job

<selenium.webdriver.remote.webelement.WebElement (session="9193aa530fdca73b480b28a428e5ea66", element="6dc01425-07b2-4364-beb2-574863565160")>

In [89]:
## Entering 'Data Scientist' in search bar
search_job.send_keys('Data Scientist')

In [90]:
search_btn=driver.find_element_by_xpath('//button[@class="btn"]')
search_btn.click()

In [101]:
## Checking the 'Delhi' location check box
loc_check=driver.find_element_by_xpath('/html/body/div[1]/div[3]/div[2]/section[1]/div[2]/div[1]/div[2]/div[3]/label/i')
loc_check.click()

In [ ]:
## Checking the desired salary range check box
salary_check=driver.find_element_by_xpath('/html/body/div[1]/div[3]/div[2]/section[1]/div[2]/div[2]/div[2]/div[2]/label/i')
salary_check.click()

In [103]:
## extracting tags having job titles
job_tags=driver.find_elements_by_xpath('//a[@class="title fw500 ellipsis"]')
job_titles=[]
for i in job_tags:
    job_titles.append(i.text)
job_titles[:10]

['Immediate Openings For DATA Scientist with 6 To 7 yrs of Experience',
 'Data Scientist',
 'Senior Data Scientist - Noida',
 'Data Scientist / Data Analyst',
 'Only Fresher / Data Scientist / Data Analyst / Analytics - MNC Jobs',
 'Senior Data Scientist',
 'Data Scientist/Data Analyst /Business Analyst',
 'Data Scientist',
 'Data Scientist - Machine Learning/NLP',
 'Data Scientist - Machine Learning/NLP']

In [104]:
## extracting all the tags having company names and saving the text in a list
company_tags=driver.find_elements_by_xpath('//a[@class="subTitle ellipsis fleft"]')
company_names=[]
for i in company_tags:
    company_names.append(i.text)
company_names[:10]

['Entune IT Consulting Private Limited',
 'Decimal Technologies Pvt Ltd.',
 'Optum Global Solutions (India) Private Limited',
 'CARS24',
 'GABA Consultancy services',
 'inVentiv International Pharma Services Pvt. Ltd.',
 'Inflexion Analytix Private Limited',
 'R Systems International Ltd.',
 'TalPro',
 'TalPro']

In [105]:
## extracting tags having locations and saving the text in a list
loc_tags=driver.find_elements_by_xpath('//li[@class="fleft grey-text br2 placeHolderLi location"]')
locations=[]
for i in loc_tags:
    locations.append(i.text)
locations[:10]

['Kolkata, Hyderabad/Secunderabad, Pune, Ahmedabad, Chennai, Bangalore/Bengaluru, Delhi / NCR, Mumbai (All Areas)\n(WFH during Covid)',
 'Gurgaon/Gurugram',
 'Noida',
 'Gurgaon/Gurugram',
 'Noida, Gurgaon/Gurugram, Delhi / NCR',
 'Hyderabad/Secunderabad, Gurgaon/Gurugram, Bangalore/Bengaluru',
 'Pune, Delhi / NCR, Mumbai (All Areas)',
 'Noida(Sector-59 Noida)',
 'Gurgaon/Gurugram',
 'Gurgaon/Gurugram']

In [106]:
## extracting all the tags having experience and saving the text in a list
exp_tags=driver.find_elements_by_xpath('//li[@class="fleft grey-text br2 placeHolderLi experience"]')
experience=[]
for i in exp_tags:
    experience.append(i.text)
experience[:10]

['5-8 Yrs',
 '1-3 Yrs',
 '2-6 Yrs',
 '1-5 Yrs',
 '0-0 Yrs',
 '3-6 Yrs',
 '0-3 Yrs',
 '5-10 Yrs',
 '2-6 Yrs',
 '2-4 Yrs']

In [107]:
## Creating a dataframe to store data for 'Data Analyst' jobs in 'Bangalore' 
DS_jobs_delhi=pd.DataFrame()

DS_jobs_delhi['Title']=job_titles[:10]
DS_jobs_delhi['Company']=company_names[:10]
DS_jobs_delhi['Location']=locations[:10]
DS_jobs_delhi['Experience']=experience[:10]

DS_jobs_delhi

Title  \
0  Immediate Openings For DATA Scientist with 6 T...   
1                                     Data Scientist   
2                      Senior Data Scientist - Noida   
3                      Data Scientist / Data Analyst   
4  Only Fresher / Data Scientist / Data Analyst /...   
5                              Senior Data Scientist   
6      Data Scientist/Data Analyst /Business Analyst   
7                                     Data Scientist   
8              Data Scientist - Machine Learning/NLP   
9              Data Scientist - Machine Learning/NLP   

                                            Company  \
0              Entune IT Consulting Private Limited   
1                     Decimal Technologies Pvt Ltd.   
2    Optum Global Solutions (India) Private Limited   
3                                            CARS24   
4                         GABA Consultancy services   
5  inVentiv International Pharma Services Pvt. Ltd.   
6                Inflexion Analytix Private Limited   
7                      R Systems International Ltd.   
8                                            TalPro   
9                                            TalPro   

                                            Location Experience  
0  Kolkata, Hyderabad/Secunderabad, Pune, Ahmedab...    5-8 Yrs  
1                                   Gurgaon/Gurugram    1-3 Yrs  
2                                              Noida    2-6 Yrs  
3                                   Gurgaon/Gurugram    1-5 Yrs  
4               Noida, Gurgaon/Gurugram, Delhi / NCR    0-0 Yrs  
5  Hyderabad/Secunderabad, Gurgaon/Gurugram, Bang...    3-6 Yrs  
6              Pune, Delhi / NCR, Mumbai (All Areas)    0-3 Yrs  
7                             Noida(Sector-59 Noida)   5-10 Yrs  
8                                   Gurgaon/Gurugram    2-6 Yrs  
9                                   Gurgaon/Gurugram    2-4 Yrs

### Q4: Write a python program to scrape data for first 10 job results for Data scientist Designation in Noida location. <br>
You have to scrape company_name, No. of days ago when job was posted, Rating of the company.<br>
This task will be done in following steps:
1. first get the webpage https://www.glassdoor.co.in/index.htm
2. Enter “Data Scientist” in “Job Title,Keyword,Company” field and enter “Noida” in “location” field.
3. Then click the search button. You will land up in the below page:
4. Then scrape the data for the first 10 jobs results you get in the above shown page.
5. Finally create a dataframe of the scraped data.<br>


Note- All of the above steps have to be done in code. No step is to be done 
manually

In [113]:
##Getting the glassdoor webpage page
url='https://www.glassdoor.co.in/member/home/index.htm'
driver.get(url)

In [1]:
from bs4 import BeautifulSoup
import requests

In [2]:
requests.get('https://www.glassdoor.co.in/member/home/index.htm')

<Response [403]>

**Unable to scrape data from 'GlassDoor' as it requires login and throws [403] response to request.**

### Q6 : Scrape data of first 100 sunglasses listings on flipkart.com. You have to 
scrape four attributes:
1. Brand
2. Product Description
3. Price
4. Discount %<br>
**To scrape the data you have to go through following steps:**<br>
- Go to flipkart webpage by url https://www.flipkart.com/
- Enter “sunglasses” in the search field where “search for products, brands and more” is written and click the search icon
- after that you will reach to a webpage having a lot of sunglasses. From this page you can scrap the required data as usual.
- after scraping data from the first page, go to the “Next” Button at the bottom of the page , then click on it
- Now scrape data from this page as usual
- repeat this until you get data for 100 sunglasses.<br>
Note that all of the above steps have to be done by coding only and not manually

In [171]:
driver=webdriver.Edge(r'C:\Users\Mandar Shinde\Desktop\DataScience\DATA TRAINED\FlipRobo Internship\msedgedriver.exe')

In [145]:
##Getting the flipkart webpage page
url='https://www.flipkart.com/'
driver.get(url)

In [146]:
## accessing the search bar and entering 'Sunglasses' in it
search_bar=driver.find_element_by_xpath('//input[@type="text"]')
search_bar.send_keys('sunglasses')

In [148]:
## Clicking the search button
search_btn=driver.find_element_by_xpath('//button[@class="L0Z3Pu"]')
search_btn.click()

In [158]:
page_urls_tags=driver.find_elements_by_xpath('//nav[@class="yFHi8N"]//a')

In [159]:
page_urls=[]
for i in page_urls_tags:
    page_urls.append(i.get_attribute('href'))
page_urls

['https://www.flipkart.com/search?q=sunglasses&otracker=search&otracker1=search&marketplace=FLIPKART&as-show=off&as=off&page=1',
 'https://www.flipkart.com/search?q=sunglasses&otracker=search&otracker1=search&marketplace=FLIPKART&as-show=off&as=off&page=2',
 'https://www.flipkart.com/search?q=sunglasses&otracker=search&otracker1=search&marketplace=FLIPKART&as-show=off&as=off&page=3',
 'https://www.flipkart.com/search?q=sunglasses&otracker=search&otracker1=search&marketplace=FLIPKART&as-show=off&as=off&page=4',
 'https://www.flipkart.com/search?q=sunglasses&otracker=search&otracker1=search&marketplace=FLIPKART&as-show=off&as=off&page=5',
 'https://www.flipkart.com/search?q=sunglasses&otracker=search&otracker1=search&marketplace=FLIPKART&as-show=off&as=off&page=6',
 'https://www.flipkart.com/search?q=sunglasses&otracker=search&otracker1=search&marketplace=FLIPKART&as-show=off&as=off&page=7',
 'https://www.flipkart.com/search?q=sunglasses&otracker=search&otracker1=search&marketplace=FLIPK

In [161]:
## Scraping the brands of sunglasses and saving them in a list
brands=[]

for i in page_urls:
    driver.get(i)
    brand_tags=driver.find_elements_by_xpath('//div[@class="_2WkVRV"]')
    for j in brand_tags:
        brands.append(j.text)
brands[:100]

['Wrogn',
 'KHALEEFA',
 'Elligator',
 'Fastrack',
 'Fastrack',
 'Fastrack',
 'kingsunglasses',
 'Villain',
 'PIRASO',
 'Fastrack',
 'kingsunglasses',
 'HAMIW COLLECTION',
 'Fastrack',
 'kingsunglasses',
 'PIRASO',
 'Villain',
 'Fastrack',
 'Fastrack',
 'Fastrack',
 'LIZA ANGEL',
 'kingsunglasses',
 'ROZZETTA CRAFT',
 'ROZZETTA CRAFT',
 'HAMIW COLLECTION',
 'PHENOMENAL',
 'Silver Kartz',
 'Singco India',
 'ROYAL SON',
 'Fastrack',
 'ROZZETTA CRAFT',
 'Elligator',
 'hipe',
 'hipe',
 'Singco India',
 'Fastrack',
 'PIRASO',
 'kingsunglasses',
 'HIPPON',
 'PIRASO',
 'ROYAL SON',
 'kingsunglasses',
 'NuVew',
 'Fastrack',
 'Wrogn',
 'PHENOMENAL',
 'NuVew',
 'hipe',
 'KHALEEFA',
 'ROZZETTA CRAFT',
 'kingsunglasses',
 'Singco',
 'HAMIW COLLECTION',
 'Fravy',
 'Fastrack',
 'PIRASO',
 'LIZA ANGEL',
 'Fastrack',
 'NuVew',
 'Fastrack',
 'HAMIW COLLECTION',
 'hipe',
 'Fastrack',
 'GANSTA',
 'ROYAL SON',
 'Fastrack',
 'DEIXELS',
 'Singco India',
 'PIRASO',
 'Villain',
 'ROYAL SON',
 'ROYAL SON',
 'hi

In [163]:
## Scraping the prices of sunglasses and saving them in a list
prices=[]

for i in page_urls:
    driver.get(i)
    price_tags=driver.find_elements_by_xpath('//div[@class="_30jeq3"]')
    for j in price_tags:
        prices.append(j.text)
prices[:100]

['₹739',
 '₹369',
 '₹295',
 '₹669',
 '₹455',
 '₹509',
 '₹299',
 '₹179',
 '₹225',
 '₹969',
 '₹210',
 '₹279',
 '₹599',
 '₹299',
 '₹225',
 '₹209',
 '₹599',
 '₹539',
 '₹949',
 '₹664',
 '₹225',
 '₹499',
 '₹404',
 '₹225',
 '₹379',
 '₹246',
 '₹203',
 '₹219',
 '₹599',
 '₹426',
 '₹331',
 '₹399',
 '₹210',
 '₹334',
 '₹609',
 '₹255',
 '₹349',
 '₹257',
 '₹225',
 '₹1,009',
 '₹225',
 '₹210',
 '₹969',
 '₹739',
 '₹379',
 '₹204',
 '₹319',
 '₹179',
 '₹474',
 '₹229',
 '₹239',
 '₹549',
 '₹269',
 '₹609',
 '₹225',
 '₹279',
 '₹669',
 '₹309',
 '₹1,479',
 '₹209',
 '₹299',
 '₹609',
 '₹211',
 '₹664',
 '₹559',
 '₹202',
 '₹331',
 '₹225',
 '₹599',
 '₹217',
 '₹474',
 '₹219',
 '₹516',
 '₹189',
 '₹398',
 '₹399',
 '₹249',
 '₹969',
 '₹398',
 '₹379',
 '₹209',
 '₹299',
 '₹239',
 '₹739',
 '₹303',
 '₹569',
 '₹265',
 '₹369',
 '₹282',
 '₹255',
 '₹499',
 '₹279',
 '₹295',
 '₹170',
 '₹399',
 '₹209',
 '₹210',
 '₹249',
 '₹569',
 '₹664']

In [164]:
## Scraping the descriptions of sunglasses and saving them in a list
description=[]

for i in page_urls:
    driver.get(i)
    description_tags=driver.find_elements_by_xpath('//a[@class="IRpwTa"]')
    for j in description_tags:
        description.append(j.text)
description[:100]

['Mirrored Wayfarer Sunglasses (51)',
 'UV Protection Round Sunglasses (Free Size)',
 'UV Protection Round Sunglasses (54)',
 'UV Protection Wayfarer Sunglasses (Free Size)',
 'Gradient, UV Protection Wayfarer Sunglasses (Free Size)',
 'UV Protection Rectangular Sunglasses (Free Size)',
 'Mirrored, UV Protection Wayfarer Sunglasses (Free Size)',
 'Others Wayfarer Sunglasses (Free Size)',
 'UV Protection Aviator Sunglasses (54)',
 'UV Protection Aviator Sunglasses (58)',
 'UV Protection Rectangular Sunglasses (Free Size)',
 'UV Protection Wayfarer Sunglasses (56)',
 'Mirrored, UV Protection Wayfarer Sunglasses (Free Size)',
 'UV Protection Aviator Sunglasses (54)',
 'UV Protection Wayfarer, Sports, Shield, Rectangular, Re...',
 'UV Protection Wayfarer Sunglasses (Free Size)',
 'UV Protection Aviator Sunglasses (Free Size)',
 'UV Protection Aviator Sunglasses (58)',
 'Polarized, UV Protection Round Sunglasses (49)',
 'UV Protection Round Sunglasses (54)',
 'UV Protection Retro Square Sun

In [165]:
## scraping the discounts and saving them in a list
discount=[]

for i in page_urls:
    driver.get(i)
    discount_tags=driver.find_elements_by_xpath('//div[@class="_3Ay6Sb"]')
    for j in discount_tags:
        discount.append(j.text)
discount[:100]

['71% off',
 '71% off',
 '88% off',
 '25% off',
 '43% off',
 '36% off',
 '88% off',
 '20% off',
 '85% off',
 '33% off',
 '25% off',
 '88% off',
 '82% off',
 '88% off',
 '85% off',
 '45% off',
 '33% off',
 '79% off',
 '32% off',
 '65% off',
 '77% off',
 '26% off',
 '30% off',
 '86% off',
 '79% off',
 '81% off',
 '83% off',
 '66% off',
 '85% off',
 '25% off',
 '70% off',
 '85% off',
 '74% off',
 '72% off',
 '32% off',
 '78% off',
 '76% off',
 '80% off',
 '81% off',
 '73% off',
 '87% off',
 '74% off',
 '83% off',
 '71% off',
 '77% off',
 '81% off',
 '80% off',
 '88% off',
 '82% off',
 '78% off',
 '76% off',
 '45% off',
 '26% off',
 '80% off',
 '32% off',
 '65% off',
 '85% off',
 '25% off',
 '66% off',
 '86% off',
 '78% off',
 '32% off',
 '78% off',
 '66% off',
 '30% off',
 '83% off',
 '83% off',
 '85% off',
 '78% off',
 '68% off',
 '20% off',
 '78% off',
 '35% off',
 '66% off',
 '72% off',
 '73% off',
 '79% off',
 '25% off',
 '81% off',
 '74% off',
 '85% off',
 '88% off',
 '80% off',
 '71

In [169]:
## Saving the scraped data in a dataframe
sunglasses_flipkart=pd.DataFrame()

sunglasses_flipkart['Brand']=brands[:100]
sunglasses_flipkart['Price']=prices[:100]
sunglasses_flipkart['Description']=description[:100]
sunglasses_flipkart['Discount']=discount[:100]

sunglasses_flipkart

Brand Price                                        Description  \
0              Wrogn  ₹739                  Mirrored Wayfarer Sunglasses (51)   
1           KHALEEFA  ₹369         UV Protection Round Sunglasses (Free Size)   
2          Elligator  ₹295                UV Protection Round Sunglasses (54)   
3           Fastrack  ₹669      UV Protection Wayfarer Sunglasses (Free Size)   
4           Fastrack  ₹455  Gradient, UV Protection Wayfarer Sunglasses (F...   
..               ...   ...                                                ...   
95  HAMIW COLLECTION  ₹209                UV Protection Round Sunglasses (53)   
96    kingsunglasses  ₹210   UV Protection Rectangular Sunglasses (Free Size)   
97    kingsunglasses  ₹249         UV Protection Retro Square Sunglasses (88)   
98         ROYAL SON  ₹569  UV Protection, Night Vision, Gradient, Mirrore...   
99         ROYAL SON  ₹664  UV Protection Wayfarer, Sports, Shield, Rectan...   

   Discount  
0   71% off  
1   71% off  
2   88% off  
3   25% off  
4   43% off  
..      ...  
95  65% off  
96  82% off  
97  81% off  
98  88% off  
99  86% off  

[100 rows x 4 columns]

### Q7: Scrape 100 reviews data from flipkart.com for iphone11 phone. You have to go the link: <br>

https://www.flipkart.com/apple-iphone-11-black-64-gb-includesearpodspoweradapter/p/itm0f37c2240b217?pid=MOBFKCTSVZAXUHGR&lid=LSTMOBFKCTSVZAXUHGREPBFGI&marketplace <br>

As shown in the above page you have to scrape the tick marked attributes.
These are 
1. Rating 
2. Review_summary 
3. Full review<br>
You have to scrape this data for first 100 reviews.

In [172]:
##Getting the desired webpage page
url='https://www.flipkart.com/apple-iphone-11-black-64-gb-includesearpodspoweradapter/p/itm0f37c2240b217?pid=MOBFKCTSVZAXUHGR&lid=LSTMOBFKCTSVZAXUHGREPBFGI&marketplace'
driver.get(url)

In [176]:
url_tags=driver.find_elements_by_xpath('//nav[@class="yFHi8N"]//a')

In [178]:
rating_urls=[]
for i in url_tags:
    rating_urls.append(i.get_attribute('href'))
rating_urls

['https://www.flipkart.com/apple-iphone-11-black-64-gb-includes-earpods-power-adapter/product-reviews/itm0f37c2240b217?pid=MOBFKCTSVZAXUHGR&lid=LSTMOBFKCTSVZAXUHGREPBFGI&marketplace=FLIPKART&page=1',
 'https://www.flipkart.com/apple-iphone-11-black-64-gb-includes-earpods-power-adapter/product-reviews/itm0f37c2240b217?pid=MOBFKCTSVZAXUHGR&lid=LSTMOBFKCTSVZAXUHGREPBFGI&marketplace=FLIPKART&page=2',
 'https://www.flipkart.com/apple-iphone-11-black-64-gb-includes-earpods-power-adapter/product-reviews/itm0f37c2240b217?pid=MOBFKCTSVZAXUHGR&lid=LSTMOBFKCTSVZAXUHGREPBFGI&marketplace=FLIPKART&page=3',
 'https://www.flipkart.com/apple-iphone-11-black-64-gb-includes-earpods-power-adapter/product-reviews/itm0f37c2240b217?pid=MOBFKCTSVZAXUHGR&lid=LSTMOBFKCTSVZAXUHGREPBFGI&marketplace=FLIPKART&page=4',
 'https://www.flipkart.com/apple-iphone-11-black-64-gb-includes-earpods-power-adapter/product-reviews/itm0f37c2240b217?pid=MOBFKCTSVZAXUHGR&lid=LSTMOBFKCTSVZAXUHGREPBFGI&marketplace=FLIPKART&page=5',


In [180]:
##extracting all tags having star rating and saving the ratings in a list
ratings=[]
for i in rating_urls:
    driver.get(i)
    rating_tags=driver.find_elements_by_xpath('//div[@class="_3LWZlK _1BLPMq"]')
    for j in rating_tags:
        ratings.append(j.text)
ratings[:100]

['5',
 '5',
 '5',
 '5',
 '5',
 '5',
 '5',
 '4',
 '5',
 '5',
 '5',
 '5',
 '5',
 '5',
 '5',
 '5',
 '5',
 '5',
 '5',
 '5',
 '4',
 '5',
 '5',
 '5',
 '5',
 '4',
 '5',
 '5',
 '5',
 '5',
 '5',
 '5',
 '5',
 '5',
 '4',
 '5',
 '5',
 '5',
 '5',
 '5',
 '5',
 '5',
 '5',
 '4',
 '5',
 '5',
 '5',
 '5',
 '5',
 '5',
 '5',
 '5',
 '5',
 '5',
 '5',
 '5',
 '5',
 '5',
 '5',
 '5',
 '5',
 '5',
 '4',
 '5',
 '5',
 '4',
 '5',
 '4',
 '5',
 '5',
 '5',
 '5',
 '5',
 '5',
 '5',
 '5',
 '5',
 '4',
 '5',
 '3',
 '5',
 '5',
 '5',
 '5',
 '5',
 '4',
 '5',
 '5',
 '5',
 '5',
 '5',
 '3',
 '5',
 '5',
 '5',
 '5',
 '5',
 '5',
 '5',
 '5']

In [182]:
## scraping the review headings and saving them in a list 
headings=[]
for i in rating_urls:
    driver.get(i)
    heading_tags=driver.find_elements_by_xpath('//p[@class="_2-N8zT"]')
    for j in heading_tags:
        headings.append(j.text)
headings[:100]

['Simply awesome',
 'Brilliant',
 'Perfect product!',
 'Fabulous!',
 'Worth every penny',
 'Great product',
 'Worth every penny',
 'Good choice',
 'Highly recommended',
 'Perfect product!',
 'Best in the market!',
 'Perfect product!',
 'Classy product',
 'Highly recommended',
 'Worth every penny',
 'Perfect product!',
 'Highly recommended',
 'Simply awesome',
 'Worth every penny',
 'Terrific',
 'Nice product',
 'Wonderful',
 'Classy product',
 'Brilliant',
 'Must buy!',
 'Good choice',
 'Perfect product!',
 'Terrific purchase',
 'Wonderful',
 'Great product',
 'Simply awesome',
 'Mind-blowing purchase',
 'Terrific purchase',
 'Simply awesome',
 'Good quality product',
 'Very poor',
 'Perfect product!',
 'Must buy!',
 'Fabulous!',
 'Mind-blowing purchase',
 'Mind-blowing purchase',
 'Wonderful',
 'Worth every penny',
 'Great product',
 'Very Good',
 'Perfect product!',
 'Classy product',
 'Perfect product!',
 'Just wow!',
 'Awesome',
 'Terrific',
 'Terrific purchase',
 'Awesome',
 'Clas

In [183]:
## Scraping the detailed reviews and saving them in a list
reviews=[]
for i in rating_urls:
    driver.get(i)
    review_tags=driver.find_elements_by_xpath('//div[@class="t-ZTKy"]')
    for j in review_tags:
        reviews.append(j.text)
reviews[:100]

['Really satisfied with the Product I received... It’s totally genuine and the packaging was also really good so if ur planning to buy just go for it.',
 'The Best Phone for the Money\n\nThe iPhone 11 offers superb cameras, a more durable design and excellent battery life for an affordable price.\n\nCompelling ultra-wide camera\nNew Night mode is excellent\nLong battery life',
 'Amazing phone with great cameras and better battery which gives you the best performance. I just love the camera .',
 'This is my first iOS phone. I am very happy with this product. Very much satisfied with this. I love this phone.',
 'Previously I was using one plus 3t it was a great phone\nAnd then I decided to upgrade I am stuck between Samsung s10 plus or iPhone 11\nI have seen the specs and everything were good except the display it’s somewhere between 720-1080 and it’s not even an amoled it’s an LCD display\nBut I decided to go with iPhone because I have never used an IOS device I have Been an android use

In [184]:
## Saving the data in dataframe
iPhone_reviews=pd.DataFrame()

iPhone_reviews['Star Rating']=ratings[:100]
iPhone_reviews['Review_Heading']=headings[:100]
iPhone_reviews['Review']=reviews[:100]

iPhone_reviews

Star Rating         Review_Heading  \
0            5         Simply awesome   
1            5              Brilliant   
2            5       Perfect product!   
3            5              Fabulous!   
4            5      Worth every penny   
..         ...                    ...   
95           5              Fabulous!   
96           5              Just wow!   
97           5  Mind-blowing purchase   
98           5              Excellent   
99           5              Excellent   

                                               Review  
0   Really satisfied with the Product I received.....  
1   The Best Phone for the Money\n\nThe iPhone 11 ...  
2   Amazing phone with great cameras and better ba...  
3   This is my first iOS phone. I am very happy wi...  
4   Previously I was using one plus 3t it was a gr...  
..                                                ...  
95  Everything is perfect pictures come out so cle...  
96  The ultimate performance\nCamera is superb\nTh...  
97  Excellent camera 📸 And Display touching very N...  
98  A perfect phone and a good battery super camer...  
99  Value for money product. This iphone 11 is rea...  

[100 rows x 3 columns]

### Q8: Scrape data for first 100 sneakers you find when you visit flipkart.com and search for “sneakers” in the search field.<br>
You have to scrape 4 attributes of each sneaker :
1. Brand
2. Product Description
3. Price
4. discount %
As shown in the below image, you have to scrape the tick marked attributes.<br>
Also note that all the steps required during scraping should be done through code 
only and not manually

In [191]:
##Getting the flipkart's webpage page
url='https://www.flipkart.com/'
driver.get(url)

In [192]:
## Accessing search bar and entering 'sneakers' in it
search_bar=driver.find_element_by_xpath('//input[@type="text"]')
search_bar.send_keys('sneakers')

In [193]:
## Clicking search button
search_btn=driver.find_element_by_xpath('//button[@class="L0Z3Pu"]')
search_btn.click()

In [194]:
page_urls_tags=driver.find_elements_by_xpath('//nav[@class="yFHi8N"]//a')

page_urls=[]
for i in page_urls_tags:
    page_urls.append(i.get_attribute('href'))
page_urls

['https://www.flipkart.com/search?q=sneakers&otracker=search&otracker1=search&marketplace=FLIPKART&as-show=on&as=off&page=1',
 'https://www.flipkart.com/search?q=sneakers&otracker=search&otracker1=search&marketplace=FLIPKART&as-show=on&as=off&page=2',
 'https://www.flipkart.com/search?q=sneakers&otracker=search&otracker1=search&marketplace=FLIPKART&as-show=on&as=off&page=3',
 'https://www.flipkart.com/search?q=sneakers&otracker=search&otracker1=search&marketplace=FLIPKART&as-show=on&as=off&page=4',
 'https://www.flipkart.com/search?q=sneakers&otracker=search&otracker1=search&marketplace=FLIPKART&as-show=on&as=off&page=5',
 'https://www.flipkart.com/search?q=sneakers&otracker=search&otracker1=search&marketplace=FLIPKART&as-show=on&as=off&page=6',
 'https://www.flipkart.com/search?q=sneakers&otracker=search&otracker1=search&marketplace=FLIPKART&as-show=on&as=off&page=7',
 'https://www.flipkart.com/search?q=sneakers&otracker=search&otracker1=search&marketplace=FLIPKART&as-show=on&as=off&p

In [195]:
## Scraping brands of first 100 sneakers and saving them in a list
brands=[]

for i in page_urls:
    driver.get(i)
    brand_tags=driver.find_elements_by_xpath('//div[@class="_2WkVRV"]')
    for j in brand_tags:
        brands.append(j.text)
brands[:100]

['Echor',
 'Labbin',
 'ASTEROID',
 'Chevit',
 'Numenzo',
 'Magnolia',
 'Numenzo',
 'DUCATI',
 'Chevit',
 'BRUTON',
 'PEHANOSA',
 'NIKE',
 'bluemaker',
 'World Wear Footwear',
 'Chevit',
 'Xylus',
 'believe',
 'RODDICK SHOES',
 'luxury fashion',
 'Strollin',
 'Chevit',
 'Shoes Bank',
 'Robbie jones',
 'DUCATI',
 'Red Rose',
 'SCATCHITE',
 'BRUTON',
 'Edoeviv',
 'Chevit',
 'Nilatin',
 'India hub',
 'PUMA',
 'World Wear Footwear',
 'Kraasa',
 'SPARX',
 'EZDEZARIO',
 'Nilatin',
 'Chevit',
 'ROCKFIELD',
 'Chevit',
 'RODDICK SHOES',
 'Fzzirok',
 'Creer',
 'DUCATI',
 'BIRDE',
 'Ktiz',
 'Edoeviv',
 'NIKE',
 'Zsyto',
 'World Wear Footwear',
 'SPARX',
 'Xylus',
 'India hub',
 'D-SNEAKERZ',
 'HOCKWOOD',
 'BRUTON',
 'PUMA',
 'World Wear Footwear',
 'SPARX',
 'DUCATI',
 'PUMA',
 'Zixer',
 'Labbin',
 'Strollin',
 'Edoeviv',
 'tigonis',
 'Jack Diamond',
 'EZDEZARIO',
 'Chevit',
 'PUMA',
 'Airland',
 'Edoeviv',
 'Labbin',
 'Robbie jones',
 'Ktiz',
 'luxury fashion',
 'Jokatoo',
 'bluemaker',
 'Zorth',

In [196]:
## Scraping prices for first 100 sneakers and saving them in a list
prices=[]

for i in page_urls:
    driver.get(i)
    price_tags=driver.find_elements_by_xpath('//div[@class="_30jeq3"]')
    for j in price_tags:
        prices.append(j.text)
prices[:100]

['₹449',
 '₹349',
 '₹499',
 '₹420',
 '₹356',
 '₹426',
 '₹378',
 '₹1,259',
 '₹399',
 '₹399',
 '₹536',
 '₹3,020',
 '₹399',
 '₹240',
 '₹474',
 '₹599',
 '₹254',
 '₹399',
 '₹679',
 '₹399',
 '₹331',
 '₹399',
 '₹474',
 '₹1,209',
 '₹299',
 '₹594',
 '₹398',
 '₹449',
 '₹499',
 '₹240',
 '₹449',
 '₹3,690',
 '₹407',
 '₹416',
 '₹629',
 '₹499',
 '₹426',
 '₹447',
 '₹260',
 '₹424',
 '₹474',
 '₹360',
 '₹398',
 '₹539',
 '₹624',
 '₹384',
 '₹474',
 '₹1,259',
 '₹378',
 '₹240',
 '₹625',
 '₹3,020',
 '₹449',
 '₹348',
 '₹674',
 '₹599',
 '₹1,470',
 '₹240',
 '₹629',
 '₹399',
 '₹2,899',
 '₹555',
 '₹399',
 '₹1,209',
 '₹449',
 '₹359',
 '₹635',
 '₹399',
 '₹348',
 '₹1,350',
 '₹251',
 '₹499',
 '₹449',
 '₹379',
 '₹429',
 '₹449',
 '₹506',
 '₹399',
 '₹449',
 '₹399',
 '₹499',
 '₹629',
 '₹1,299',
 '₹539',
 '₹279',
 '₹346',
 '₹378',
 '₹449',
 '₹267',
 '₹699',
 '₹499',
 '₹1,259',
 '₹474',
 '₹378',
 '₹398',
 '₹3,020',
 '₹1,115',
 '₹311',
 '₹1,322',
 '₹599']

In [197]:
## Scraping discounts for first 100 sneakers and saving them in a list
discount=[]

for i in page_urls:
    driver.get(i)
    discount_tags=driver.find_elements_by_xpath('//div[@class="_3Ay6Sb"]')
    for j in discount_tags:
        discount.append(j.text)
discount[:100]

['46% off',
 '55% off',
 '75% off',
 '71% off',
 '64% off',
 '78% off',
 '62% off',
 '65% off',
 '88% off',
 '60% off',
 '66% off',
 '45% off',
 '60% off',
 '51% off',
 '52% off',
 '62% off',
 '74% off',
 '69% off',
 '72% off',
 '60% off',
 '66% off',
 '60% off',
 '52% off',
 '68% off',
 '70% off',
 '77% off',
 '60% off',
 '43% off',
 '80% off',
 '51% off',
 '85% off',
 '53% off',
 '68% off',
 '58% off',
 '10% off',
 '50% off',
 '67% off',
 '75% off',
 '47% off',
 '78% off',
 '52% off',
 '63% off',
 '60% off',
 '46% off',
 '68% off',
 '61% off',
 '40% off',
 '65% off',
 '70% off',
 '51% off',
 '16% off',
 '45% off',
 '85% off',
 '47% off',
 '32% off',
 '62% off',
 '57% off',
 '51% off',
 '16% off',
 '88% off',
 '42% off',
 '44% off',
 '60% off',
 '68% off',
 '55% off',
 '80% off',
 '36% off',
 '60% off',
 '30% off',
 '59% off',
 '49% off',
 '50% off',
 '55% off',
 '62% off',
 '66% off',
 '43% off',
 '66% off',
 '60% off',
 '55% off',
 '69% off',
 '81% off',
 '10% off',
 '75% off',
 '55

In [198]:
## Scraping short descriptions for first 100 sneakers and saving them in a list
description=[]

for i in page_urls:
    driver.get(i)
    description_tags=driver.find_elements_by_xpath('//a[@class="IRpwTa"]')
    for j in description_tags:
        description.append(j.text)
description[:100]

['Raptor Sneakers For Men',
 'Sneakers For Men',
 "Original Luxury Branded Fashionable Men's Casual Walkin...",
 'Unique & Perfect Collection Combo Pack of 02 Shoes for ...',
 'Sneakers For Men',
 'Sneakers For Men',
 'Sneakers For Men',
 'Sneakers For Men',
 'Combo Pack Of 4 Casual Shoes Loafer Shoes Sneakers For ...',
 'casual for men (blue 06) Sneakers For Men',
 'Super Stylish & Trendy Combo Pack of 02 Pairs Sneakers ...',
 'Sneakers For Men',
 'Sneakers For Men',
 '5011-Latest Collection Stylish Casual Loafer Sneakers S...',
 'Fashion Outdoor Canvas Casual Light Weight Lace-up Even...',
 'Sneakers For Men',
 '494 Perfect Sports Shoes for Running Training Hikking &...',
 'Luxury Fashionable casual sneaker shoes Sneakers For Me...',
 'Speed Set of 5 Pairs Sneakers Outdoors Casuals for Men ...',
 "Luxury Branded Fashionable Men's Casual Walking Partywe...",
 "White Sneaker For Men's/Boy's Sneakers For Men",
 'Sneakers for men(black_6) Sneakers For Men',
 'Sneakers For Men',
 'MAPF1 X

In [199]:
## Saving the scraped data in a dataframe
sneakers_flipkart=pd.DataFrame()

sneakers_flipkart['Brand']=brands[:100]
sneakers_flipkart['Description']=description[:100]
sneakers_flipkart['Price']=prices[:100]
sneakers_flipkart['Discount']=discount[:100]

sneakers_flipkart

Brand                                        Description   Price  \
0            Echor                            Raptor Sneakers For Men    ₹449   
1           Labbin                                   Sneakers For Men    ₹349   
2         ASTEROID  Original Luxury Branded Fashionable Men's Casu...    ₹499   
3           Chevit  Unique & Perfect Collection Combo Pack of 02 S...    ₹420   
4          Numenzo                                   Sneakers For Men    ₹356   
..             ...                                                ...     ...   
95  luxury fashion                           Max IDP Sneakers For Men  ₹3,020   
96            PUMA  Chevit Latest Fashion Combo Pack of 2 Pairs Ca...  ₹1,115   
97        Magnolia                                   Sneakers For Men    ₹311   
98            PUMA                                   Sneakers For Men  ₹1,322   
99          DUCATI  Fashionable casual sneaker shoes Sneakers For Men    ₹599   

   Discount  
0   46% off  
1   55% off  
2   75% off  
3   71% off  
4   64% off  
..      ...  
95  62% off  
96  62% off  
97  68% off  
98  66% off  
99  88% off  

[100 rows x 4 columns]

### Q9: Go to the link - https://www.myntra.com/shoes . Set Price filter to “Rs. 6649 to Rs. 13099” , Color filter to “Black”, as shown in  the below image<br>

And then scrape First 100 shoes data you get. The data should include “Brand” of 
the shoes , Short Shoe description, price of the shoe as shown in the below image<br>



Please note that applying the filter and scraping the data , everything should be 
done through code only and there should not be any manual step.

In [200]:
##Getting the Myntra's webpage page
url='https://www.myntra.com/shoes'
driver.get(url)

In [201]:
## Checking price range checkbox
price_check=driver.find_element_by_xpath('/html/body/div[2]/div/div[1]/main/div[3]/div[1]/section/div/div[5]/ul/li[2]/label/div')
price_check.click()

In [202]:
## Checking 'Black' checkbox
color_check=driver.find_element_by_xpath('/html/body/div[2]/div/div[1]/main/div[3]/div[1]/section/div/div[6]/ul/li[1]/label/div')
color_check.click()

In [205]:
## Scraping urls for all pages where black shoes are listed and saving them in a list
page_urls_tags=driver.find_elements_by_xpath('//ul[@class="pagination-container"]//a')

page_urls=[]
for i in page_urls_tags:
    page_urls.append(i.get_attribute('href'))
page_urls

['https://www.myntra.com/shoes?f=Color%3ABlack_36454f&plaEnabled=false&rf=Price%3A6392.0_12595.0_6392.0%20TO%2012595.0',
 'https://www.myntra.com/shoes?f=Color%3ABlack_36454f&plaEnabled=false&rf=Price%3A6392.0_12595.0_6392.0%20TO%2012595.0&p=2',
 'https://www.myntra.com/shoes?f=Color%3ABlack_36454f&plaEnabled=false&rf=Price%3A6392.0_12595.0_6392.0%20TO%2012595.0&p=3',
 'https://www.myntra.com/shoes?f=Color%3ABlack_36454f&plaEnabled=false&rf=Price%3A6392.0_12595.0_6392.0%20TO%2012595.0&p=4',
 'https://www.myntra.com/shoes?f=Color%3ABlack_36454f&plaEnabled=false&rf=Price%3A6392.0_12595.0_6392.0%20TO%2012595.0&p=5',
 'https://www.myntra.com/shoes?f=Color%3ABlack_36454f&plaEnabled=false&rf=Price%3A6392.0_12595.0_6392.0%20TO%2012595.0&p=6',
 'https://www.myntra.com/shoes?f=Color%3ABlack_36454f&plaEnabled=false&rf=Price%3A6392.0_12595.0_6392.0%20TO%2012595.0&p=7',
 'https://www.myntra.com/shoes?f=Color%3ABlack_36454f&plaEnabled=false&rf=Price%3A6392.0_12595.0_6392.0%20TO%2012595.0&p=8',
 'ht

In [206]:
## Scraping brands of first 100 shoes and saving them in a list
brands=[]

for i in page_urls:
    driver.get(i)
    brand_tags=driver.find_elements_by_xpath('//h3[@class="product-brand"]')
    for j in brand_tags:
        brands.append(j.text)
brands[:100]

['ALDO',
 'Nike',
 'Geox',
 'UNDER ARMOUR',
 'Puma',
 'Geox',
 'UNDER ARMOUR',
 'UNDER ARMOUR',
 'UNDER ARMOUR',
 'UNDER ARMOUR',
 'UNDER ARMOUR',
 'ADIDAS',
 'UNDER ARMOUR',
 'UNDER ARMOUR',
 'Puma',
 'Puma',
 'UNDER ARMOUR',
 'UNDER ARMOUR',
 'UNDER ARMOUR',
 'UNDER ARMOUR',
 'Puma',
 'UNDER ARMOUR',
 'Geox',
 'UNDER ARMOUR',
 'UNDER ARMOUR',
 'Hush Puppies',
 'Hush Puppies',
 'FORCLAZ By Decathlon',
 'Skechers',
 'PUMA Hoops',
 'UNDER ARMOUR',
 'Hush Puppies',
 'Puma',
 'Quechua By Decathlon',
 'Cole Haan',
 'UNDER ARMOUR',
 'Puma',
 'Geox',
 'PUMA Hoops',
 'Geox',
 'RARE RABBIT',
 'Puma',
 'Geox',
 'RARE RABBIT',
 'Nike',
 'Geox',
 'UNDER ARMOUR',
 'Geox',
 'RARE RABBIT',
 'Cole Haan',
 'Geox',
 'Cole Haan',
 'Geox',
 'Geox',
 'RARE RABBIT',
 'Nike',
 'J.FONTINI',
 'Bugatti',
 'Geox',
 'UNDER ARMOUR',
 'Hush Puppies',
 'Hush Puppies',
 'DAVINCHI',
 'Geox',
 'Geox',
 'Ruosh',
 'Geox',
 'Cole Haan',
 'Ruosh',
 'Geox',
 'Puma',
 'J.FONTINI',
 'Ruosh',
 'J.FONTINI',
 'Bugatti',
 'RARE 

In [207]:
## Scraping prices of first 100 shoes and saving them in a list
prices=[]

for i in page_urls:
    driver.get(i)
    price_tags=driver.find_elements_by_xpath('//div[@class="product-price"]')
    for j in price_tags:
        prices.append(j.text)
prices[:100]

['Rs. 11999',
 'Rs. 8999',
 'Rs. 11436Rs. 17595(35% OFF)',
 'Rs. 7773Rs. 8449(8% OFF)',
 'Rs. 6999',
 'Rs. 10999',
 'Rs. 10999',
 'Rs. 9999',
 'Rs. 7999',
 'Rs. 6999',
 'Rs. 6999Rs. 9999(30% OFF)',
 'Rs. 6999',
 'Rs. 6999',
 'Rs. 10999',
 'Rs. 9749Rs. 14999(35% OFF)',
 'Rs. 7199Rs. 8999(20% OFF)',
 'Rs. 7499',
 'Rs. 10999',
 'Rs. 7999',
 'Rs. 7999',
 'Rs. 7999',
 'Rs. 6499',
 'Rs. 11999',
 'Rs. 8999Rs. 9999(10% OFF)',
 'Rs. 9999',
 'Rs. 6999Rs. 9999(30% OFF)',
 'Rs. 6999',
 'Rs. 6499',
 'Rs. 8999',
 'Rs. 7499',
 'Rs. 8099Rs. 8999(10% OFF)',
 'Rs. 6499',
 'Rs. 8799Rs. 10999(20% OFF)',
 'Rs. 9999',
 'Rs. 8799Rs. 10999(20% OFF)',
 'Rs. 9999',
 'Rs. 10999',
 'Rs. 10490',
 'Rs. 7149Rs. 10999(35% OFF)',
 'Rs. 8449Rs. 12999(35% OFF)',
 'Rs. 6999',
 'Rs. 11999',
 'Rs. 6499',
 'Rs. 9999',
 'Rs. 8121Rs. 12495(35% OFF)',
 'Rs. 9999',
 'Rs. 8399Rs. 10499(20% OFF)',
 'Rs. 6499',
 'Rs. 9999',
 'Rs. 7999Rs. 9999(20% OFF)',
 'Rs. 9999',
 'Rs. 7999Rs. 9999(20% OFF)',
 'Rs. 10399Rs. 12999(20% OFF)',
 'R

In [208]:
## Scraping descriptions for first 100 shoes and saving them in a list
description=[]

for i in page_urls:
    driver.get(i)
    description_tags=driver.find_elements_by_xpath('//h4[@class="product-product"]')
    for j in description_tags:
        description.append(j.text)
description[:100]

['Men Driving Shoes',
 'LEBRON XVIII Basketball Shoes',
 'Men Charged Bandit 6 Running',
 'Men Cell Fraction Fade Running',
 'Men Leather Formal Derbys',
 'Women HOVR Sonic 4 Colorshift',
 'Men Textured Leather Loafers',
 'Men HOVR Sonic 4 Colorshift',
 'Unisex Project Rock Recruit',
 'Men Charged Vantage Running',
 'Men Charged Pursuit 2 SE Shoes',
 'Women SUPERNOVA Running Shoes',
 'Men Charged Assert 8 Marble',
 'Men Charged Pursuit 2 SE',
 'Men Velocity Nitro Running',
 'Women Deviate Nitro Running',
 'GS SC 3Zero IV Basketball',
 'Men HOVR Sonic 4 Running Shoes',
 'Men Printed Slip-On Sneakers',
 'Men Charged Rogue 2.5 RFLCT',
 'Men Charged Commit 3 Training',
 'Women Charged Vantage ClrShft',
 'Men SOFTRIDE Rift Shoes',
 'Men HOVR Infinite Running',
 'Men HOVR Sonic STRT Tech Shoes',
 'Men Formal Derbys',
 'Men Driving Shoes',
 'Men Leather Slip-On Sneakers',
 'TREKKING 100 Boots',
 'Men Equalizer 4.0 Trail',
 'Unisex Clyde Basketball Shoes',
 'Men Semi-Formal Slip-Ons',
 'Women 

In [209]:
## Saving the scraped data in a dataframe
black_shoes=pd.DataFrame()
black_shoes['Brand']=brands[:100]
black_shoes['Description']=description[:100]
black_shoes['Price']=prices[:100]

black_shoes

Brand                          Description  \
0           ALDO                    Men Driving Shoes   
1           Nike        LEBRON XVIII Basketball Shoes   
2           Geox         Men Charged Bandit 6 Running   
3   UNDER ARMOUR       Men Cell Fraction Fade Running   
4           Puma            Men Leather Formal Derbys   
..           ...                                  ...   
95          Geox  Men GENERATION ZEROGRAND STITCHLITE   
96     Cole Haan               Women AIR ZOOM Running   
97          Nike         Men Leather Slip-On Sneakers   
98          Geox            Men Leather Driving Shoes   
99          Geox                Men Textured Sneakers   

                          Price  
0                     Rs. 11999  
1                      Rs. 8999  
2   Rs. 11436Rs. 17595(35% OFF)  
3      Rs. 7773Rs. 8449(8% OFF)  
4                      Rs. 6999  
..                          ...  
95                     Rs. 9490  
96                     Rs. 9990  
97   Rs. 7039Rs. 10999(36% OFF)  
98    Rs. 6399Rs. 7999(20% OFF)  
99    Rs. 6399Rs. 7999(20% OFF)  

[100 rows x 3 columns]

### Q10: Go to webpage https://www.amazon.in/ . Enter “Laptop” in the search field and then click the search icon.
Then set CPU Type filter to “Intel Core i7” and “Intel Core i9” as shown in the 
below image:

After setting the filters scrape first 10 laptops data. You have to scrape 3 attributes 
for each laptop:
1. title
2. Ratings
3. Price
As shown in the below image as the tick marked attributes

In [34]:
driver=webdriver.Edge(r'C:\Users\Mandar Shinde\Desktop\DataScience\DATA TRAINED\FlipRobo Internship\msedgedriver.exe')

In [35]:
##Getting the amazon's webpage page
url='https://www.amazon.in/'
driver.get(url)

In [36]:
## accessing the search bar and entering 'Laptop' in it
search_bar=driver.find_element_by_xpath('/html/body/div[1]/header/div/div[1]/div[2]/div/form/div[2]/div[1]/input')
search_bar.send_keys('Laptop')

In [37]:
## Clicking the search button
search_btn=driver.find_element_by_xpath('/html/body/div[1]/header/div/div[1]/div[2]/div/form/div[3]/div/span/input')
search_btn.click()

In [38]:
## Checking the i7 processor checkbox
i7_check=driver.find_element_by_xpath('/html/body/div[1]/div[2]/div[1]/div/div[2]/div/div[3]/span/div[1]/span/div/div/div[5]/ul[5]/li[26]/span/a/div/label/i')
i7_check.click()

In [39]:
## Checking the i9 processor checkbox
i9_check=driver.find_element_by_xpath('/html/body/div[1]/div[2]/div[1]/div/div[2]/div/div[3]/span/div[1]/span/div/div/div[5]/ul[5]/li[28]/span/a/div/label/i')
i9_check.click()

In [40]:
##extracting tags which have titles and saving the titles in a list
title_tags=driver.find_elements_by_xpath('//h2[@class="a-size-mini a-spacing-none a-color-base s-line-clamp-2"]')
titles=[]
for i in title_tags:
    titles.append(i.text)
titles[:10]

['ASUS ZenBook 13 OLED (2021) Intel Core i7-1165G7 11th Gen 14" (35.56cms) FHD Thin and Light Laptop (16GB/1TB SSD/Windows 10/Office 2019/Intel Iris Xᵉ Graphics/Lilac Mist/1.14 kg), UX325EA-KG701TS',
 'ASUS VivoBook S S15 (2021), 15.6-inch (39.62 cms) FHD Intel Core i7-1165G7 11th Gen, Thin and Light Laptop (8GB/512GB SSD/Office 2019/Windows 10/2GB NVIDIA MX350 Graphics/Silver/1.8 kg) S532EQ-BQ702TS',
 'MSI GF65 Thin, Intel i7-10750H, 15.6" FHD (39.6 cm) IPS-Level 144Hz Panel Laptop (16GB/512GB NVMe SSD/Windows 10 Home/Nvidia GTX1660 Ti 6GB GDDR6/Black/1.86Kg), 10SDR-1280IN',
 'HP Pavilion (2021) Thin & Light 11th Gen Core i7 Laptop, 16 GB RAM, 1TB SSD, Iris Xe Graphics, 14" (35.56cms) FHD Screen, Windows 10, MS Office, Backlit Keyboard (14-dv0058TU)',
 'Mi Notebook Horizon Edition 14 Intel Core i7-10510U 10th Gen 14-inch (35.56 cms) Thin and Light Laptop(8GB/512GB SSD/Windows 10/Nvidia MX350 2GB Graphics/Grey/1.35Kg), XMA1904-AF+Webcam',
 'ASUS TUF Gaming F15 (2020), 15.6-inch (39.62 

In [41]:
##extracting tags which have prices and saving the prices in a list
price_tags=driver.find_elements_by_xpath('//span[@class="a-price"]')
prices=[]
for i in price_tags:
    prices.append(i.text)
prices[:10]

['₹1,02,990',
 '₹83,595',
 '₹81,990',
 '₹84,990',
 '₹59,490',
 '₹71,990',
 '₹34,990',
 '₹74,990',
 '₹90,490',
 '₹1,09,990']

In [50]:
url_tags=driver.find_elements_by_xpath('//div[@class="a-section a-spacing-none"]//a')
urls=[]
for i in url_tags:
    urls.append(i.get_attribute('href'))
urls[:10]


['https://www.amazon.in/MSI-i7-10750H-IPS-Level-Windows-10SDR-1280IN/dp/B093L8JCZV/ref=sr_1_1_sspa?dchild=1&keywords=Laptop&qid=1628793935&refinements=p_n_feature_thirteen_browse-bin%3A12598163031%7C16757432031&rnid=12598141031&s=computers&sr=1-1-spons&psc=1&spLa=ZW5jcnlwdGVkUXVhbGlmaWVyPUE2UVROWFAyRzdGU08mZW5jcnlwdGVkSWQ9QTAzOTk1MTkxOUw1Mk5TUVdMNDZRJmVuY3J5cHRlZEFkSWQ9QTAxODczOTQxRjhMNThUVFU1T1Q2JndpZGdldE5hbWU9c3BfYXRmJmFjdGlvbj1jbGlja1JlZGlyZWN0JmRvTm90TG9nQ2xpY2s9dHJ1ZQ==#',
 'https://www.amazon.in/MSI-i7-10750H-IPS-Level-Windows-10SDR-1280IN/dp/B093L8JCZV/ref=sr_1_1_sspa?dchild=1&keywords=Laptop&qid=1628793935&refinements=p_n_feature_thirteen_browse-bin%3A12598163031%7C16757432031&rnid=12598141031&s=computers&sr=1-1-spons&psc=1&spLa=ZW5jcnlwdGVkUXVhbGlmaWVyPUE2UVROWFAyRzdGU08mZW5jcnlwdGVkSWQ9QTAzOTk1MTkxOUw1Mk5TUVdMNDZRJmVuY3J5cHRlZEFkSWQ9QTAxODczOTQxRjhMNThUVFU1T1Q2JndpZGdldE5hbWU9c3BfYXRmJmFjdGlvbj1jbGlja1JlZGlyZWN0JmRvTm90TG9nQ2xpY2s9dHJ1ZQ==#',
 'javascript:void(0)',
 'https:/

In [51]:
import time

In [55]:
ratings=[]

for i in urls:
    driver.get(i)
    time.sleep(5)
    try:
        rating=driver.find_element_by_xpath('//span[@class="a-size-base a-nowrap"]//span')
        ratings.append(rating.text)
    except:
        ratings.append('---')
ratings

WebDriverException: Message: unknown error: unsupported protocol
  (Session info: MicrosoftEdge=92.0.902.67)


In [48]:
driver.find_element_by_xpath('//span[@class="a-size-base a-nowrap"]//span').text

'3.7 out of 5'

### Unable to fetch ratings for computers

In [20]:
### Saving the scraped data in a dataframe
import pandas as pd
laptops=pd.DataFrame()

laptops['Title']=titles[:10]
laptops['Price']=prices[:10]


laptops

Title      Price
0  HP Pavilion (2021) Thin & Light 11th Gen Core ...    ₹84,990
1  Acer Swift 5 14" (35.56cms) Full HD IPS Displa...    ₹94,365
2  HP Pavilion (2021) Thin & Light 11th Gen Core ...    ₹84,990
3  Life Digital Laptop 15.6-inch (39.62 cms) (Int...    ₹29,990
4  Mi Notebook Horizon Edition 14 Intel Core i7-1...    ₹59,999
5  ASUS TUF Gaming F15 (2020), 15.6-inch (39.62 c...    ₹71,990
6  NEXSTGO Primus NX201 NP15N1IN008P 15.6-inch La...    ₹56,155
7  Life Digital Laptop 15.6-inch (39.62 cms) (Int...    ₹34,990
8  OMEN 17 Gaming Laptop, NVIDIA GeForce RTX 2070...    ₹74,990
9  MSI GF75 Thin, Intel i7-10750H, 17.3" (43.9 cm...  ₹1,05,990